### DISCUSSION TITLE
Use MongoDB session/transaction in collection operations

### DISCUSSION THREAD

**Discussion by languitar** [quarkusio/quarkus#27289] [Q&A] [answered]
I use MongoDB Panache with the transaction support. This works well for most cases. However, in a few cases I have to resort to using a `MongoCollection` instance directly. For these cases, is there any way to access the MongoDB session used by panache to also use an ongoing transaction in my custom collection methods?

[a comment by an automated account is not shown]

**Reply by loicmathieu** [reply]
There sure be a hacky way to obtain it, I can give you the few lines ;)

But as it may be of interest for multiple person, and as I think it's a legitimate need better open an enhancement request so we provide a way to access the session inside a transaction (or a least document it).

**Reply by languitar** [reply]
I've seen the hacky way to get an active session, but when in need of creating a new session because my code is the first one to deal with a new transaction, afterwards registering the session for panache operations would even be a lot more hacky.

I have opened a request as proposed: https://github.com/quarkusio/quarkus/issues/27311

**Reply by loicmathieu** [reply]
When getting the active session we can create it if needed, this will solves the issue with new session.
Let's continue the discussion in the enhancement issue.
Can you mark this discussion as close ?

**Reply by languitar** [reply]
I can mark the thread opened by the bot as the answer. That sounds a bit funny, though ;-) Is that what you mean?

**Reply by loicmathieu** [reply]
Oh yes, sorry it's my fault, I reply on the bot answer and not on a dedicated one ;)
Let's mark it as answer anyway.
